File: retort-core/src/main/java/co/kaioru/retort/annotation/IAnnotationProcessor.java
package co.kaioru.retort.annotation;

import co.kaioru.retort.ICommand;
import co.kaioru.retort.ICommandContext;

import java.lang.reflect.Method;

public interface IAnnotationProcessor<I extends ICommandContext, O> extends IAnnotationAdapter<I, O> {

    void process(ICommand<I, O> command, Object object, Method method);

}

File: retort-annotations/src/main/kotlin/co/kaioru/retort/annotation/IAnnotationProcessor.kt
package co.kaioru.retort.annotation

import co.kaioru.retort.ICommand
import co.kaioru.retort.annotation.exceptions.AnnotationAdapterException
import java.lang.reflect.Method

interface IAnnotationProcessor<I, O> : IAnnotationAdapter<I, O> {
    @Throws(AnnotationAdapterException::class)
    fun process(command: ICommand<I, O>, any: Any, method: Method)
}